scripts/model_gen_ng/model_gen_ng/templates/enum.tpl

```diff
@@ -0,0 +1,60 @@
+namespace [[ enum.namespace ]] {
+    [[ enum.doc | docstring ]]
+    enum Values
+    [%- if enum.underlying_type is defined %] : [[ enum.underlying_type ]][% endif %]{
+        Invalid,
+        [% for member in enum.members %]
+        [[- member.doc | docstring ]]
+        [[- member.name ]],
+        [% endfor %]
+        Count
+    };
+
+    inline llvm::StringRef getName(Values V) {
+        switch (V) {
+            case Invalid:
+                return "Invalid";
+            [%- for member in enum.members %]
+            case [[ member.name ]]:
+                return "[[ member.name ]]";
+            [%- endfor %]
+            default:
+                revng_abort();
+                break;
+        }
+    }
+
+    inline Values fromName(llvm::StringRef Name) {
+        [%- for member in enum.members %]
+        [%- if loop.first %]
+        if (Name == "[[ member.name ]]") {
+            return [[ member.name ]];
+        }
+        [%- else %]
+        else if (Name == "[[ member.name ]]") {
+            return [[ member.name ]];
+        }
+        [%- endif -%]
+        [%- endfor -%]
+        else {
+            return Invalid;
+        }
+    }
+
+    [[ enum.customcode ]]
+} // namespace [[ enum.namespace ]]
+
+template<>
+struct KeyedObjectTraits<[[ enum.fullname ]]>
+  : public IdentityKeyedObjectTraits<[[ enum.fullname ]]> {};
+
+namespace llvm::yaml {
+template<>
+struct ScalarEnumerationTraits<[[ enum.fullname ]]>
+  : public NamedEnumScalarTraits<[[ enum.fullname ]]> {
+  static constexpr const char Name[] = "[[ enum.name ]]";
```

Comment:
We already have this in `TupleLikeTraits`.

Reply:
But enums are not TupleLikes?

Reply:
Yeah, but they're kinda part of the family. The alternative is to make a new trait, these are for YAML stuff.

Reply:
Updates on this matter?

Reply:
I'd say just add a `constexpr StringRef name() { return "asdf"; }` in the `enum` namespace and use that.

We can switch to a trait if we need to. It'd look something like:

```c++
// In some header
template<typename Enum>
StringRef getEnumName();
```
```c++
// In the generated code for the enum
template<>
StringRef getEnumName<MyEnum>() { return "MyEnum"; }
```